app/src/main/java/com/darshan/notificity/NotificationsActivity.kt

```
@@ -98,11 +103,43 @@ fun NotificationSearchScreen(
+    fun handleClearAll() {
+        isClearingAll = true
+    }
+
+
+    // Added: Launch effect to clear notifications after animation
+    LaunchedEffect(isClearingAll) {
+        if (isClearingAll) {
+            try {
+                kotlinx.coroutines.delay(400) // Match animation duration
+                clearAllNotifications()
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)